tests/page-objects/default/common/modal.wdio.page.js

```diff
@@ -32,10 +32,21 @@ const cancel = async (timeout) => {
   await checkModalHasClosed();
 };
 
+const hideOverlay = () => {
```

Comment:
I know this is cheap and hacky. It feels like I tried everything else except for checking if the modal exists before every action.

Reply:
If this is what we have to do then so be it...

It would be nice if the refresh was called in the sync method to guarantee this was reverted.

I think it should also be made a private function inside common.wdio because we don't want it to be used widely, IMO.

Reply:
I moved it to a private function. 
Sync does automatically refresh when we expect an update. Otherwise, I wouldn't overcomplicate the code and tests and force refreshes.